src/main/java/org/prebid/server/bidder/ix/IxBidder.java

```diff
@@ -97,84 +102,85 @@ public IxBidder(String endpointUrl, JacksonMapper mapper) {
         return Result.of(httpRequests, errors);
     }
 
-    private static void validateImp(Imp imp) {
-        if (imp.getBanner() == null) {
-            throw new PreBidException(String.format("Invalid MediaType. Ix supports only Banner type. "
-                    + "Ignoring ImpID=%s", imp.getId()));
+    private ExtImpIx parseImpExt(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), IX_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage(), e);
         }
     }
 
-    private void makeRequests(BidRequest bidRequest, Imp imp, List<BidRequest> prioritizedRequests,
-                              List<BidRequest> regularRequests) {
-        List<Format> formats = imp.getBanner().getFormat();
-        if (CollectionUtils.isEmpty(formats)) {
-            formats = makeFormatFromBannerWidthAndHeight(imp);
-        }
-        final ExtImpIx extImpIx = parseAndValidateImpExt(imp);
-        final List<BidRequest> modifiedBidRequests = createBidRequest(bidRequest, imp, formats, extImpIx);
-        if (!modifiedBidRequests.isEmpty()) {
-            if (modifiedBidRequests.size() == 1) {
-                prioritizedRequests.addAll(modifiedBidRequests);
-            } else {
-                prioritizedRequests.add(modifiedBidRequests.get(0));
-                regularRequests.addAll(modifiedBidRequests.subList(1, modifiedBidRequests.size()));
-            }
+    private static List<BidRequest> makeBidRequests(BidRequest bidRequest,
+                                                    Imp imp,
+                                                    ExtImpIx extImpIx) {
+        final ArrayList<BidRequest> modifiedBidRequests = new ArrayList<>();
+        final Site modifiedSite = modifiedSite(bidRequest.getSite(), extImpIx);
+
+        final List<Imp> modifiedImps = modifiedImps(imp);
+        for (Imp modifiedImp : modifiedImps) {
+            final BidRequest modifiedBidRequest = bidRequest.toBuilder()
+                    .site(modifiedSite)
+                    .imp(Collections.singletonList(modifiedImp))
+                    .build();
+            modifiedBidRequests.add(modifiedBidRequest);
         }
+
+        return modifiedBidRequests;
     }
 
-    private static List<Format> makeFormatFromBannerWidthAndHeight(Imp imp) {
-        final Banner banner = imp.getBanner();
-        return Collections.singletonList(
-                Format.builder().w(banner.getW()).h(banner.getH()).build());
+    private static Site modifiedSite(Site site, ExtImpIx extImpIx) {
```

Comment:
i'd suggest to name functions in imperative mood, like `modifySite(..)`
Pls check other fun-s below if agree.